### PR TITLE
Fix non-float32 efficientnet calls

### LINKED ISSUE #16401: EfficentNets no longer train on non-float32 inputs
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- TensorFlow installed from (source or binary): Pip `tf-nightly`
- TensorFlow version (use command below): 2.10.0-dev20220409
- Python version: 3.7.3
- GPU model and memory: N/A - reproducible on CPU, GPU, and TPU

**Describe the problem**.
[This PR](https://github.com/keras-team/keras/pull/16032/files) merged to solve [an issue with efficientnet normalisation](https://github.com/tensorflow/tensorflow/issues/49930) hardcoded a set of float values into the efficientnet architecture.

These magic numbers are fixed to float32 when running in graph mode, and will raise an exception as soon as the network is called on float16 or bfloat16 inputs. [bfloat16 inputs are recommended when training on TPUs](https://cloud.google.com/tpu/docs/bfloat16#improving_performance_with_bfloat16) 

**Describe the current behavior**.
An exception is raised whenever a keras efficientnet is trained on a non-float32 input

**Describe the expected behavior**.
An exception is not raised whenever a keras efficientnet is trained on a non-float32 input

- Do you want to contribute a PR? (yes/no): yes
- Briefly describe your candidate solution(if contributing):  Replace the hardcoded graph constants with:

`x = layers.Rescaling(1. / tf.math.sqrt(IMAGENET_STDDEV_RGB))(x)`

**Standalone code to reproduce the issue**.

```python
import tensorflow as tf
from tensorflow.keras.applications import efficientnet as efn

tf.keras.mixed_precision.set_global_policy("mixed_bfloat16")

ds = (
    tf.data.Dataset.from_tensor_slices(
        (
            tf.ones([1, 224, 224, 3], dtype=tf.bfloat16),
            tf.ones([1, 1000], dtype=tf.float32),
        )
    )
    .repeat()
    .batch(batch_size=1)
)

e = efn.EfficientNetB0()
e.compile(loss="binary_crossentropy")
e.fit(ds, steps_per_epoch=1, epochs=1)
```

### PR DESCRIPTION
**Pull request by hctomkins**
Fix #16401 

No obvious way I could see to include a breaking example of the old pattern in tests, so left as-is given it's a one-liner.

### CODE REVIEW

**Review by fchollet — COMMENTED** [review]
Thanks for the PR. Are you able to run the unit tests locally?

**Comment by fchollet** [comment]
I'm actually concerned about "There were tests whose specified size is too big"

Can you try to run the test "keras/applications/applications_load_weight_test.py" manually?

**Comment by hctomkins** [comment]
@fchollet this passes:
```
keras-vscode@11b39a2d6d36:/home/keras$ bazel test keras/applications:applications_load_weight_test_resnet
WARNING: The following configs were expanded more than once: [v2]. For repeatable flags, repeats are counted twice and may lead to unexpected behavior.
INFO: Options provided by the client:
  Inherited 'common' options: --isatty=1 --terminal_columns=116
INFO: Reading rc options for 'test' from /home/keras/.bazelrc:
  Inherited 'build' options: --apple_platform_type=macos --define open_source_build=true --define=use_fast_cpp_protos=false --define=tensorflow_enable_mlir_generated_gpu_kernels=0 --define=allow_oversize_protos=true --spawn_strategy=standalone -c opt --announce_rc --define=grpc_no_ares=true --config=short_logs --config=v2
INFO: Reading rc options for 'test' from /home/keras/.bazelrc:
  'test' options: --define open_source_build=true --define=use_fast_cpp_protos=false --config=v2
INFO: Found applicable config definition build:short_logs in file /home/keras/.bazelrc: --output_filter=DONT_MATCH_ANYTHING
INFO: Found applicable config definition build:v2 in file /home/keras/.bazelrc: --define=tf_api_version=2 --action_env=TF2_BEHAVIOR=1
INFO: Found applicable config definition build:v2 in file /home/keras/.bazelrc: --define=tf_api_version=2 --action_env=TF2_BEHAVIOR=1
INFO: Analyzed target //keras/applications:applications_load_weight_test_resnet (0 packages loaded, 1 target configured).
INFO: Found 1 test target...
Target //keras/applications:applications_load_weight_test_resnet up-to-date:
  bazel-bin/keras/applications/applications_load_weight_test_resnet
INFO: Elapsed time: 50.240s, Critical Path: 49.71s
INFO: 5 processes: 4 internal, 1 local.
INFO: Build completed successfully, 5 total actions
//keras/applications:applications_load_weight_test_resnet                PASSED in 49.6s

INFO: Build completed successfully, 5 total actions
```

(fyi the test name has a suffix _resnet in the `BUILD` file - `applications_load_weight_test.py` is still the target)

**Review by fchollet — APPROVED** [review]
Ok, thanks for checking. LGTM
